### ISSUE TITLE
A couple of typos should be fixed

### ISSUE THREAD

**Issue by martinbis11** [isocpp/CppCoreGuidelines#556]
A couple of typos should be fixed.  Here they are with the rules they come from.
### ES.23 Prefer the {} initializer syntax

The Fibonacci series in the example is wrong...

```
auto fib10 = {0, 1, 2, 3, 5, 8, 13, 25, 38, 63};   // fib10 is a list
```

should be:

```
auto fib10 = {0, 1, 2, 3, 5, 8, 13, 21, 34, 55};   // fib10 is a list
```
### ES.71: Prefer a range-for-statement to a for-statement when there is a choice

The example:

```
for (int i = 0; i < v.size(); ++i) { // body messes with loop variable: can't be a range-for
    if (i % 2)
        ++i;   // skip even elements
    else
        cout << v[i] << '\n';
}
```

has a bug.  The "++i;" should actually be a "continue;" statement, or an empty statement.  Incrementing it in the loop body will result in a double increment, and only one print statement (for i == 0) will happen.  Also, if the even elements are to be skipped, the test should be:

```
 if (i % 2 == 0)
```
### ES.75: Avoid do-statements

The example:

```
int x;
do {
    cin >> x;
    x
} while (x < 0);
```

has an extra "x" that should not be there.
### E.6: Use RAII to prevent leaks

> We are on a system so small that the exception support would eat up most of our 2K or memory.

should read "our 2K of memory.".
### E.27: If you can't throw exceptions, use error codes systematically

> to gather cleanup at the end of the function to avoid repetittion:

should read 'repetition'.
### Con.3: By default, pass pointers and references to consts

> an object passed by pointer or reference to non-cost

should read 'non-const'.
### T.40: Use function objects to pass operations to algorithms

> You can, of course, gneralize those functions 

should read 'generalize'.

Also, in the example:

```
auto y1 = find_if(v, [](Ordered x) { return x>7; }); // reruire an ordered type
```

should read 'require'.
### T.61: Do not over-parameterize members (SCARY)

In the example, the following member is declared:

```
Node* head;
```

However, it should probably be the declared internal structure:

```
Link* head;
```
### T.81: Do not mix hierarchies and arrays

In the second example, we are using the "aa" variable, but it should be "va".
### If a class is a container, give it an initializer-list constructor

In the example:

```
vector<std::initializer_list<T>>;
```

should probably read:

```
Vector(std::initializer_list<T>);
```
### Never let a pointer outlive the object it points to

The example uses the following phrase:

```
vector<string> v = { "this", "will", "cause" "trouble" };
```

However, in the second example:

```
string x = *p;   // undefined behavior: x may not be "this"
```

Using "this" between quotes like this might be confusing since "this" is a C++ keyword.  A different example should avoid such reserved keywords just to remove ambiguity.

**Comment by AndrewPardoe** [comment]
Martyn, we agree with these changes. Have you prepared a PR?

**Comment by severinmeyer** [comment]
As a note, with the merge of pull request #804, all these typos reported by @martinbis11 are now fixed.

**Comment by gdr-at-ms** [comment]
Fixed by #804
